public/pages/workflows/new_workflow/new_workflow.tsx

```diff
@@ -72,25 +148,72 @@ export function NewWorkflow(props: NewWorkflowProps) {
   // initial hook to populate all workflows
   // enrich them with dynamically-generated UI flows based on use case
   useEffect(() => {
-    if (presetWorkflows) {
-      setAllWorkflows(
-        presetWorkflows.map((presetWorkflow) =>
-          enrichPresetWorkflowWithUiMetadata(presetWorkflow)
-        )
-      );
-    }
-  }, [presetWorkflows]);
+    const loadWorkflows = async () => {
+      if (!presetWorkflows || presetWorkflows.length === 0) {
+        return;
+      }
 
-  // initial hook to populate filtered workflows
-  useEffect(() => {
-    setFilteredWorkflows(allWorkflows);
-  }, [allWorkflows]);
+      const dataSourceEnabled = getDataSourceEnabled().enabled;
+
+      if (!dataSourceEnabled) {
+        const enrichedWorkflows = presetWorkflows.map((presetWorkflow) =>
+          enrichPresetWorkflowWithUiMetadata(
+            presetWorkflow,
+            MINIMUM_FULL_SUPPORTED_VERSION
+          )
+        );
+        setAllWorkflows(enrichedWorkflows);
+        setFilteredWorkflows(enrichedWorkflows);
+        setIsVersionLoading(false);
+        return;
+      }
+
+      if (!dataSourceId) {
+        setAllWorkflows([]);
+        setFilteredWorkflows([]);
+        setIsVersionLoading(true);
+        return;
+      }
+
+      setIsVersionLoading(true);
+
+      try {
```

Comment:
Same as previous comment, why do we need a try/catch here?

Reply:
make sense. Addressed on line 172 by removing the unnecessary try-catch-block since the logic is handled in the getVersion and filter function.